services/data-info/src/data_info/services/trash.clj

```diff
@@ -69,10 +71,13 @@
         ;;; If the file isn't already in the user's trash, move it there
         ;;; otherwise, do a hard delete.
         (if-not (.startsWith p (paths/user-trash-path user))
-          (move-to-trash cm p user)
+          (do (let [trash-path (move-to-trash cm p user)]
+              (reset! trash-paths
+                      (assoc @trash-paths p trash-path))))
```

Comment:
Ah, good idea. I'd just copied the way this code works from lower in the file for the restore functionality, which I'll probably change too.